Core/AppURLs+StatisticsDependent.swift

```diff
@@ -0,0 +1,148 @@
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import Foundation
+import BrowserServicesKit
+
+public extension URL {
+
+    private static let defaultStatisticsDependentURL = StatisticsDependentURL()
+
+    static func makeSearchURL(text: String) -> URL? { defaultStatisticsDependentURL.makeSearchURL(text: text) }
+    static func makeSearchURL(query: String, queryContext: URL? = nil) -> URL? {
+        defaultStatisticsDependentURL.makeSearchURL(query: query, queryContext: queryContext)
+    }
+    static func applyingStatsParams(for url: URL) -> URL { defaultStatisticsDependentURL.applyingStatsParams(for: url) }
+    static var searchAtb: URL? { defaultStatisticsDependentURL.searchAtb }
+    static var appAtb: URL? { defaultStatisticsDependentURL.appAtb }
+    static func hasCorrectMobileStatsParams(url: URL) -> Bool { defaultStatisticsDependentURL.hasCorrectMobileStatsParams(url: url) }
+    static func makePixelURL(pixelName: String, formFactor: String? = nil, includeATB: Bool = true) -> URL {
+        defaultStatisticsDependentURL.makePixelURL(pixelName: pixelName, formFactor: formFactor, includeATB: includeATB)
+    }
+
+}
+
+public final class StatisticsDependentURL {
+
+    private let statisticsStore: StatisticsStore
+
+    init(statisticsStore: StatisticsStore = StatisticsUserDefaults()) {
+        self.statisticsStore = statisticsStore
+    }
+
+    // MARK: - Search
+
+    func makeSearchURL(text: String) -> URL? {
+        makeSearchURL(text: text, additionalParameters: [])
+    }
+
+    func makeSearchURL(query: String, queryContext: URL? = nil) -> URL? {
+        if let url = URL.webUrl(from: query) {
+            return url
+        }
+
+        var parameters = [String: String]()
+        if let queryContext = queryContext,
+           queryContext.isDuckDuckGoSearch,
+           queryContext.getParameter(named: URL.Param.verticalMaps) == nil,
+           let vertical = queryContext.getParameter(named: URL.Param.vertical),
+           URL.ParamValue.majorVerticals.contains(vertical) {
+
+            parameters[URL.Param.verticalRewrite] = vertical
+        }
+
+        return makeSearchURL(text: query, additionalParameters: parameters)
+    }
+
+    /**
+     Generates a search url with the source (t) https://duck.co/help/privacy/t
+     and cohort (atb) https://duck.co/help/privacy/atb
+     */
+    private func makeSearchURL<C: Collection>(text: String, additionalParameters: C) -> URL
+    where C.Element == (key: String, value: String) {
+        let searchURL = URL.ddg
+            .appendingParameter(name: URL.Param.search, value: text)
+            .appendingParameters(additionalParameters)
+        return applyingStatsParams(for: searchURL)
+    }
+
+    func applyingStatsParams(for url: URL) -> URL {
```

Comment:
super-nit: I think I would prefer using `to` rather than `for` if trying to be grammatical, but I think you can simply remove the parameter name altogether since this is really internal to this file